File: src/agile-chat-web/src/types/Tool.ts
export type ToolStatus = 'Active' | 'Inactive' | 'Deprecated';

export type MethodType = 'GET' | 'POST' ;

export type Tool = {
  id: string;
  name: string;
  type: 'Database' | 'LogicApp' | 'ExternalAPI';
  status: ToolStatus;
  description?: string;
  createddate :  string;
  lastupdateddate : string;
  jsonTemplate: string;
  databaseDSN: string;
  databaseQuery: string;
  method: MethodType;
  api: string;
};

 

File: src/agile-chat-api/Models/Tool.cs
using agile_chat_api.Enums;
using System.Text.Json.Serialization;

public enum ToolType
{
    Database,
    LogicApp,
    ExternalAPI
}

public class Tool
{
    public Guid id { get; set; } = Guid.NewGuid();
    
    public string Name { get; set; } = string.Empty;
    
    public string type { get; set; } = string.Empty;

    [JsonConverter(typeof(JsonStringEnumConverter))]
    public ToolStatus Status { get; set; } = ToolStatus.Active;
    
    public string? Description { get; set; }

    public string JsonTemplate { get; set; } = string.Empty;

    public string DatabaseDSN { get; set; } = string.Empty;

    public string Method { get; set; } = string.Empty;
    public string Api { get; set; }= string.Empty;
    public string DatabaseQuery { get; set; }= string.Empty;
    public string CreatedDate { get; set; }  = string.Empty;
    public string LastUpdatedDate { get; set; }  = string.Empty;
}
